File: DTOs/Sources/PollServiceDTOs/GetPollDTO.swift
import Foundation

public struct GetPollDTO: Codable {
    public var id: UUID
    public var question: String
    public var description: String
    public var startedAt: Date
    public var closedAt: Date
    public var anonymous: Bool
    public var multiSelect: Bool
    public var iVoted: Bool
    public var isOpen: Bool
    public var options: [GetPollVotingOptionDTO]
    
    public init(id: UUID, question: String, description: String, startedAt: Date, closedAt: Date, anonymous: Bool, multiSelect: Bool, iVoted: Bool, isOpen: Bool, options: [GetPollVotingOptionDTO]) {
        self.id = id
        self.question = question
        self.description = description
        self.startedAt = startedAt
        self.closedAt = closedAt
        self.anonymous = anonymous
        self.multiSelect = multiSelect
        self.iVoted = iVoted
        self.isOpen = isOpen
        self.options = options
    }
}

File: android/app/src/main/java/net/ipv64/kivop/dtos/PollServiceDTOs/GetPollDTO.kt
// MIT No Attribution
// 
// Copyright 2025 KIVoP
// 
// Permission is hereby granted, free of charge, to any person obtaining a copy of this
// software and associated documentation files (the Software), to deal in the Software
// without restriction, including without limitation the rights to use, copy, modify,
// merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
// permit persons to whom the Software is furnished to do so.
// 
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

package net.ipv64.kivop.dtos.PollServiceDTOs

import java.util.UUID
import java.time.LocalDateTime

data class GetPollDTO (
    var id : UUID,
    var question : String,
    var description : String,
    var startedAt : LocalDateTime,
    var closedAt : LocalDateTime,
    var anonymous : Boolean,
    var multiSelect : Boolean,
    var iVoted : Boolean,
    var isOpen : Boolean,
    var options : List<GetPollVotingOptionDTO>,
)
